capabilityreceiver/pkg/capabilityhandler/capabilityreceiver.go

```diff
@@ -168,30 +173,57 @@ func saveMsgInDB(message []byte, args *map[string]interface{}) error {
 	} else if err != nil {
 		return err
 	} else {
-		// If the endpoint info does exist, update it with the new information.
-		existingEndpt.CapabilityStatement = fhirEndpoint.CapabilityStatement
-		existingEndpt.TLSVersion = fhirEndpoint.TLSVersion
-		existingEndpt.MIMETypes = fhirEndpoint.MIMETypes
-		existingEndpt.HTTPResponse = fhirEndpoint.HTTPResponse
-		existingEndpt.Errors = fhirEndpoint.Errors
-		existingEndpt.Validation = fhirEndpoint.Validation
-		existingEndpt.SMARTHTTPResponse = fhirEndpoint.SMARTHTTPResponse
-		existingEndpt.SMARTResponse = fhirEndpoint.SMARTResponse
-		existingEndpt.IncludedFields = fhirEndpoint.IncludedFields
-		existingEndpt.SupportedResources = fhirEndpoint.SupportedResources
-		existingEndpt.ResponseTime = fhirEndpoint.ResponseTime
-		err = chplmapper.MatchEndpointToVendor(ctx, existingEndpt, store)
+
+		err = chplmapper.MatchEndpointToVendor(ctx, fhirEndpoint, store)
 		if err != nil {
 			return err
 		}
-		err = chplmapper.MatchEndpointToProduct(ctx, existingEndpt, store, fmt.Sprintf("%v", (*args)["chplMatchFile"]))
+
+		err = chplmapper.MatchEndpointToProduct(ctx, fhirEndpoint, store, fmt.Sprintf("%v", (*args)["chplMatchFile"]))
 		if err != nil {
 			return err
 		}
 
-		err = store.UpdateFHIREndpointInfo(ctx, existingEndpt)
-		if err != nil {
-			return err
+		existingEndpt.Metadata.URL = fhirEndpoint.Metadata.URL
+		existingEndpt.Metadata.HTTPResponse = fhirEndpoint.Metadata.HTTPResponse
+		existingEndpt.Metadata.Errors = fhirEndpoint.Metadata.Errors
+		existingEndpt.Metadata.ResponseTime = fhirEndpoint.Metadata.ResponseTime
+		existingEndpt.Metadata.SMARTHTTPResponse = fhirEndpoint.Metadata.SMARTHTTPResponse
+
+		if !existingEndpt.EqualExcludeMetadata(fhirEndpoint) {
+			// If the endpoint info does exist, update it with the new information.
+			existingEndpt.CapabilityStatement = fhirEndpoint.CapabilityStatement
+			existingEndpt.TLSVersion = fhirEndpoint.TLSVersion
+			existingEndpt.MIMETypes = fhirEndpoint.MIMETypes
+			existingEndpt.Validation = fhirEndpoint.Validation
+			existingEndpt.SMARTResponse = fhirEndpoint.SMARTResponse
+			existingEndpt.IncludedFields = fhirEndpoint.IncludedFields
+			existingEndpt.SupportedResources = fhirEndpoint.SupportedResources
+
+			err = chplmapper.MatchEndpointToVendor(ctx, existingEndpt, store)
```

Comment:
Why are MatchEndpointToVendor/Product called again on the existingEndpt?

Reply:
This is actually something I was a little confused about while doing this task and would really appreciate both of your inputs on. So one of the issues I saw while trying to test the selective update is that the existing endpoint and the new endpoint would never actually equal each other since the new endpoint did not have all the vendor information that the existing endpoint should have. Therefore I realized I would need to populate the new endpoint with the correct vendor/product information, so that is why I ended up doing this. But maybe I can just set the vendor and product info in the new fhir endpoint to the info from the existing fhir endpoint since this information I am guessing doesn't change? And if it is the case that this information doesn't change, I guess I wouldn't need to call the vendor/product matching with the existing fhir endpoint!

Reply:
That makes sense, thanks for the clarification!